File: L-16.c

#include "stdio.h"
#include "string.h"
int main()
{
    char x[100][19] = {""}, h[11] = {'1','0','x','9','8','7','6','5','4','3','2'};
    int n, y[100], f[17] = {7,9,10,5,8,4,2,1,6,3,7,9,10,5,8,4,2},out,count=0;
    scanf("%d",&n);
    for (size_t i = 0; i < n; i++)
    {
        scanf("%s",x[i]);
    }
    for (size_t i = 0; i < n; i++)
    {
        out = 0;
        int w;
        for (size_t j = 0; j < 17; j++)
        {
            w = x[i][j] - 48;
            if (w>=10||w<0)
            {
                w = -1;
                break;
            }
            else
            {
                out += w * f[j];
            }
        }
        if (w == -1)
        {
            y[count] = i;
            count++;
        }
        else
        {
            out %= 11;
            if (h[out]==x[i][17])
            {
                continue;
            }
            else
            {
                y[count] = i;
                count++;
            }
        }
    }
    if (count==0)
    {
        printf("All passed\n");
    }
    else
    {
        for (size_t i = 0; i < count; i++)
        {
            printf("%s\n",x[y[i]]);
        }
    }
    return 0;
}

File: L-16.py
#------------------------------------------部分正确--------------------------------------------

l=[7,9,10,5,8,4,2,1,6,3,7,9,10,5,8,4,2]

li=['1','0','X','9','8','7','6','5','4' ,'3','2']
b=[]
n=int(input())
x=[]
for i in range(n):
    x.append(input())
for i in range(n):
    t=0
    if 'X' in x[i][:-1]:
        b.append(i)
        continue
    for j in range(17):
        t+=l[j]*int(x[i][j])
    t%=11
    t=li[t]
    if t!=x[i][-1]:
        b.append(i)
le=len(b)
if le==0:
    print('All passed')
else:

    for i in range(le):
        print(x[b[i]])
